### PR TITLE
[mccas] Use PaddingRef for debug sections

### PR DESCRIPTION
**Pull request by felipepiovezan**
This sequence of patches makes it so that we stop ignoring PaddingRef nodes for the debug info/abbrev sections. To accomplish this, we re-root the DIE blocks as children of the DebugInfoSectionRef block, as well as delete some dead code.

### CODE REVIEW

**Comment by felipepiovezan** [comment]
@swift-ci test

File: llvm/lib/MC/MCCASObjectV1.cpp [hunk 1]
```diff
@@ -127,14 +127,13 @@ Error MCSchema::serializeObjectFile(cas::ObjectProxy RootNode,
...
-Expected<SmallVector<cas::ObjectRef>>
-loadReferences(const cas::ObjectProxy &Proxy) {
+SmallVector<cas::ObjectRef> loadReferences(const cas::ObjectProxy &Proxy) {
   SmallVector<cas::ObjectRef> Refs;
   if (auto E = Proxy.forEachReference([&](cas::ObjectRef ID) -> Error {
         Refs.push_back(ID);
         return Error::success();
       }))
-    return std::move(E);
+    llvm_unreachable("Callback never returns an error");
```
Comment (rastogishubham) [inline llvm/lib/MC/MCCASObjectV1.cpp:136]: Hehe this is a good catch

**Review by rastogishubham — APPROVED** [review]
LGTM! Nothing too crazy happening here

**Comment by felipepiovezan** [comment]
@swift-ci test

**Pull request merged**